app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -559,7 +559,10 @@ private boolean handleSocksProxyPreferenceChange(Preference preference, String n
             if (newValue.equals("")) {
                 preference.setSummary(getString(R.string.do_not_use_proxy) + " " + getString(R.string.generic_example) + ": " + getString(R.string.socks_proxy_address_example));
                 return true;
-            } else if (newValue.matches("^socks5://(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}):(\\d{1,5})$")) {
+            } else if (newValue.matches("^socks5://" +
+                                        "(((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))|" +
+                                        "((([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])))" +
+                                        ":(\\d{1,5})$")) {
```

Comment:
Can you just not check anything here?

Reply:
Whats wrong with a RFC compliant check? Don't be scared...

Reply:
or just socks5://.*:\d{1,}

Reply:
yeah that was my first guess... mandatory or optional for merge ;)?